app/models/solidus_stripe/payment_method.rb

```diff
@@ -70,12 +67,12 @@ def stripe_dashboard_url(intent_id)
       end
     end
 
-    private
+    def assign_slug
```

Comment:
I think responsibilities were fairer when that method was part of the `SlugEntry` class, especially considering that the payment method already deals with a lot of stuff (via inheritance from upstream Solidus).

Reply:
The main reason was (I think) having access to `preferred_test_mode` (see the other comment), personally I'm fine with both arrangements, with a preference for SlugEntry not needing to know too much about the payment method.